core/src/main/java/com/netflix/iceberg/BaseTableScan.java

```diff
@@ -231,4 +234,69 @@ public String toString() {
         .add("filter", rowFilter)
         .toString();
   }
+
+  private CloseableIterable<FileScanTask> planSplits(long splitSize) {
+    final CloseableIterable<FileScanTask> fileScanTasks = planFiles();
+    final Iterable<FileScanTask> splitTasks = FluentIterable
+        .from(fileScanTasks)
+        .transformAndConcat(input -> getSplits(input, splitSize));
+    // Capture manifests which can be closed after scan planning
+    final CloseableIterable<FileScanTask> closeableSplitTasks = CloseableIterable
+        .combine(splitTasks, ImmutableList.of(fileScanTasks));
+    return closeableSplitTasks;
+  }
+
+  private static Iterable<FileScanTask> getSplits(FileScanTask fileScanTask, long splitSize) {
+    return () -> new Iterator<FileScanTask>() {
+      long offset = 0;
+      long remainingLen = fileScanTask.length();
+
+      @Override
+      public boolean hasNext() {
+        return remainingLen > 0;
+      }
+
+      @Override
+      public FileScanTask next() {
+        long len = Math.min(splitSize, remainingLen);
+        final FileScanTask splitTask = splitTask(offset, len, fileScanTask);
+        offset += len;
+        remainingLen -= len;
+        return splitTask;
```

Comment:
The logic here looks reasonable to me, but adding this feature should require some tests to make sure the splits and file formats work. Basically something like this:
* Write an Avro or Parquet file that is ~64MB (with Parquet row group size = 16MB)
* Set split size to 16MB and plan splits
* Read the each task separately but in order and validate that all the rows are present

The Parquet one would be in the iceberg-parquet module.

Reply:
+1. Also, what are your thoughts on existing tests that are breaking?

Reply:
What do you mean? Tests are passing for your last commit.

Reply:
I've commented specific asserts under `TestFilteredScan` which rely on the assumption  that each task processes a single file. These changes are part of the PR.

Reply:
Okay, I'll have to take a look. Could you un-comment the tests so that we can see the failures?

Reply:
@rdblue . I've updated the PR with the tests uncomment. I have kept the following line commented
```byId.updateProperties().set("read.split.target-size", "1").commit();```
as this would have create too many small tasks.

I've also addressed rest of the comments, except adding new unit tests. Which I'll work on soon.